CodeiumVS/InlineDiff/InlineDiffAdornment.cs

```diff
@@ -0,0 +1,721 @@
+{
+    [Export(typeof(AdornmentLayerDefinition))]
+    [Name("CodeiumInlineDiffAdornment")]
+    [Order(After = "Text")]
+    private static readonly AdornmentLayerDefinition _codeiumInlineDiffAdornment;
+
+    private static readonly LineTransform _defaultTransform   = new(1.0);
+    private static MethodInfo? _fnSetInterceptsAggregateFocus = null;
+
+    private readonly IAdornmentLayer   _layer;
+    private readonly IVsTextView       _vsHostView;
+    private readonly IOleCommandTarget _nextCommand;
+
+    private InlineDiffView?            _adornment = null;
+
+    private ITrackingSpan?             _leftTrackingSpan;
+    private ITrackingSpan?             _rightTrackingSpan;
+    private ITrackingSpan?             _trackingSpanExtended;
+
+    private IProjectionBuffer?         _leftProjectionBuffer;
+    private IProjectionBuffer?         _rightProjectionBuffer;
+
+    private IReadOnlyRegion?           _leftReadOnlyRegion;
+
+    private ITextBuffer?               _rightSourceBuffer;
+    private IVsWindowFrame?            _rightWindowFrame;
+    private ITextDocument?             _rightTextDocument;
+
+    private double _adornmentTop       = 0;
+    private double _codeBlockHeight    = 0;
+
+    private LineTransform _lineTransform = _defaultTransform;
+    private ITagAggregator<InterLineAdornmentTag>? _tagAggregator = null;
+
+    public bool HasAdornment => _adornment != null;
+    public bool IsAdornmentFocused => HasAdornment && (_adornment.ActiveView != null) && _adornment.ActiveView.VisualElement.IsKeyboardFocused;
+
+    public InlineDiffAdornment(IWpfTextView view) : base(view)
+    {
+        _vsHostView = _hostView.ToIVsTextView();
+        _layer = _hostView.GetAdornmentLayer("CodeiumInlineDiffAdornment");
+
+        _hostView.Closed                += HostView_OnClosed;
+        _hostView.LayoutChanged         += HostView_OnLayoutChanged;
+        _hostView.ViewportLeftChanged   += HostView_OnLayoutChanged;
+        _hostView.ViewportWidthChanged  += HostView_OnLayoutChanged;
+        _hostView.ZoomLevelChanged      += HostView_OnZoomLevelChanged;
+        _hostView.Caret.PositionChanged += HostView_OnCaretPositionChanged;
+
+        _vsHostView.AddCommandFilter(this, out _nextCommand);
+
+        // attempt to get the `AggregateFocusInterceptor.GetInterceptsAggregateFocus`
+        // method in the Microsoft.VisualStudio.Text.Internal.dll
+        if (_fnSetInterceptsAggregateFocus == null)
+        {
+            try
+            {
+                string name = "Microsoft.VisualStudio.Text.Internal";
+                Assembly assembly = AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(assembly => assembly.GetName().Name == name);
+
+                Type type = assembly?.GetType("Microsoft.VisualStudio.Text.Editor.AggregateFocusInterceptor");
+                _fnSetInterceptsAggregateFocus = type?.GetMethod("SetInterceptsAggregateFocus", BindingFlags.Static | BindingFlags.Public);
```

Comment:
Try to get the internal `SetInterceptsAggregateFocus` method using reflection, rather than adding a reference to the assembly.